### DISCUSSION TITLE
Individual API rules for Fields

### DISCUSSION THREAD

**Discussion by ysnt-yes** [pocketbase/pocketbase#1350] [Q&A]
I know it's not 1.0 yet but I wanted to request that we get some field-specific API rules.

For example: everyone can see "profile" information IE a bio, links, and username but can't see created, updated, id, etc.

I know it's possible to use a user collection and a profile collection that have references to each other but I feel like it becomes a tangled web pretty quick.

**Comment by ganigeorgiev** [comment]
This was requested previously but for now it is left aside (it is in my todo to research it) because it'll complicate a lot of things and I have some concerns about the performance if we apply the individual field rules in the same manner as the others.

The requested feature is also a little ambiguous because it is not clear when this rule(s) will be applied - only on response serialization? What about when creating or updating the fields - should this always be allowed or it will depend on the "hidden" field rule?

_I'll move the issue to discussion. Suggestions for better handling, other possible approaches or examples are welcome._

**Reply by ganigeorgiev** [reply]
Alternatively, this may be better left to be handled, as suggested in https://github.com/pocketbase/pocketbase/issues/311#issuecomment-1312790397, by the eventual "View" collection type (I'll consider reordering the roadmap after v0.12-13 and the UI relation picker improvements).

**Reply by ysnt-yes** [reply]
Ah ok, makes sense. I don't know the first thing about Go so I wouldn't have the slightest idea on when to implement the check.
What I already know, and what I intended it to be was something like django or .net where I think it's:
request > auth > check > query > respond

or something like having multiple people can edit BUT only the owner/creator can modify certain values
